cmd/process.go

```diff
@@ -224,10 +219,17 @@ func StartProcessor(
 				msg.Updated = time.Now()
 			}
 			State.Mu.Unlock()
-
 		}
+
+		// requeue txs, ensure not to exceed retry limit
 		if requeue {
-			processingQueue <- tx
+			if dequeuedTx.RetryAttempt < cfg.Circle.FetchRetries {
+				dequeuedTx.RetryAttempt++
+				time.Sleep(time.Duration(cfg.Circle.FetchRetryInterval) * time.Second)
```

Comment:
can cfg.Circle.FetchRetryInterval ever be 0? Maybe some validation to ensure this is not the case on config load as a check. Or if 0 actually set to 1 or something

Reply:
Good catch, just added a check to ensure the circle configuration is valid. The app will exit early if anything is invalid. 

I will need to add additional validation checks to the other config options, but I'm going to save that for another PR.